frontend/src/metabase/metadata/utils/database.ts

```diff
@@ -0,0 +1,23 @@
+import type { Database } from "metabase-types/api";
+
+export function hasDatabaseFeature(
+  database: Database,
+  feature: string | null | undefined,
```

Comment:
Please use the type form `metabase-types/api`

```suggestion
  feature: DatabaseFeature | "join" | null | undefined,
```

Reply:
Unfortunately `DatabaseFeature` does not contain all the features, so this give some TS errors.
We can do this though: a8c7de36243a221b2f3da23e2ad624c38f9d5575